static/js/controllers/reports.js

```diff
@@ -131,34 +187,60 @@ var _ = require('underscore'),
           return DB.get()
             .get(report)
             .then(FormatDataRecord);
-        } else {
-          return FormatDataRecord(report);
         }
+        return FormatDataRecord(report);
       };
 
       $scope.refreshReportSilently = function(report) {
-        _fetchFormattedReport(report)
+        return _fetchFormattedReport(report)
           .then(function(doc) {
-              _setSelected(doc[0]);
-            })
+            _setSelected(doc[0]);
+          })
           .catch(function(err) {
             $log.error('Error fetching formatted report', err);
           });
       };
 
+      $scope.deselectReport = function(report) {
+        for (var i = 0; i < $scope.selected.length; i++) {
+          if ($scope.selected[i]._id === report._id) {
+            $scope.selected.splice(i, 1);
+          }
+        }
+        $('#reports-list li[data-record-id="' + report._id + '"] input[type="checkbox"]')
+          .prop('checked', false);
+        $scope.settingSelected(true);
+        setActionBar();
+      };
+
+      $scope.handleDeletedReport = function(report) {
+        if ($scope.selectMode) {
+          // remove just this one item
+          liveList.remove(report);
+          $scope.deselectReport(report);
+        } else {
+          // clear all
+          $scope.selectReport();
+        }
+      };
+
       $scope.selectReport = function(report) {
         if (!report || !liveList.initialised()) {
           $scope.clearSelected();
           return;
         }
-
-        $scope.clearSelected();
         $scope.setLoadingContent(report);
-
+        if (!$scope.selectMode) {
+          // in selected mode we append to the list so don't clear it
+          $scope.clearSelected();
+        }
         _fetchFormattedReport(report)
+          .then(function(formatted) {
+            return formatted && formatted.length && formatted[0];
```

Comment:
just `if(long condition)` in the `then` block below would be more concise (but this works too)